nef Jekyll

miguelangel-dev opened the pull request:
###  References
**Issues:**
* Close #14  
* Close #8 

In this PR we completed the render for Jekyll' documentation. 

1. `nef-jekyll` to render Jekyll website.
2. updated `nef` to supports render Jekyll website from playgrounds - connected `nef-jekyll` script to `nef`

Commit:
remove unuseful folder when build playground project

Commit:
structure for nef-doc

Find only workspace

Commit:
find playgrounds and its pages

Commit:
get playground/page name from path

miguelangel-dev committed:
fix how to get playgrounds paths

miguelangel-dev committed:
fix problem multiple spaces in playground’s pages

remove unuseful code

miguelangel-dev committed:
added `output` argument where to render Jekyll site

miguelangel-dev committed:
build jekyll files from playgrounds

miguelangel-dev committed:
fix bug - remove current directory

miguelangel-dev committed:
sidebar creation

miguelangel-dev committed:
jekyll generator page name folders - lowercased and without white spaces

miguelangel-dev committed:
trimming whitespaces in jekyll’s header file

miguelangel-dev committed:
supports to define main page

miguelangel-dev committed:
rename to nef jekyll

miguelangel-dev committed:
be more explicit with type of errors - building playgrounds

fix podfile path

miguelangel-dev committed:
support to generate playgrounds giving Podfile and bow’s branch

miguelangel-dev committed:
update `nef` to support nef-jekyll

miguelangel-dev committed:
added documentation to `nef-jekyll`

truizlop commented on bin/nef-playground:
```diff
@@ -22,29 +22,61 @@ printHelp() {
-    echo "    ${lime}${bold}version${reset}${normal} bow's version to use [default: $latestVersion]"
+    echo "    ${lime}${bold}--name${reset}${normal} create a playground project using this name [default: $DEFAULT_PLAYGROUND]"
     echo ""
+    echo "    ${lime}${bold}--bow-version${reset}${normal} bow's version to use [default: $latestVersion]"
```
```suggestion
    echo "    ${lime}${bold}--bow-version${reset}${normal} Bow's version to use [default: $latestVersion]"
```

truizlop commented on bin/nef-playground:
```diff
@@ -22,29 +22,61 @@ printHelp() {
+    echo "    ${lime}${bold}--name${reset}${normal} create a playground project using this name [default: $DEFAULT_PLAYGROUND]"
     echo ""
+    echo "    ${lime}${bold}--bow-version${reset}${normal} bow's version to use [default: $latestVersion]"
+    echo "    ${lime}${bold}--bow-branch${reset}${normal} bow's branch to use [optional]"
```
```suggestion
    echo "    ${lime}${bold}--bow-branch${reset}${normal} Bow's branch to use [optional]"
```

truizlop commented on bin/nef-playground:
```diff
@@ -22,29 +22,61 @@ printHelp() {
+  local podfile="$4"       # parameter `podfile`
+  local podfilePath="$projectFolder/Podfile"
+
+  echo -ne "${normal}Updating ${green}bow "
```
```suggestion
  echo -ne "${normal}Updating ${green}Bow "
```

truizlop commented on bin/nef:
```diff
@@ -23,20 +24,34 @@ reset=$(tput sgr0)
-    echo "    ${lime}${bold}version${reset}${normal} bow's version to use [optional]."
+    echo "    ${green}${bold}$PLAYGROUND${reset}${normal} build a playground with Bow compatibility"
+    echo "    ${lime}${bold}--name${reset}${normal} create a playground project using this name [optional]"
+    echo "    ${lime}${bold}--version${reset}${normal} bow's version to use [optional]"
```
```suggestion
    echo "    ${lime}${bold}--version${reset}${normal} Bow's version to use [optional]"
```

truizlop commented on bin/nef-playground:
```diff
@@ -22,29 +22,61 @@ printHelp() {
+#   setBowVersion(String projectFolder, String version, String branch, String podfile)
 #   - Parameter `projectFolder`: path to the project folder.
-#   - Parameter `version`: bow version which update Podfile.
+#   - Parameter `version`: bow's version which update Podfile.
```
```suggestion
#   - Parameter `version`: Bow's version which update Podfile.
```

truizlop commented on bin/nef-playground:
```diff
@@ -22,29 +22,61 @@ printHelp() {
 #   - Parameter `projectFolder`: path to the project folder.
-#   - Parameter `version`: bow version which update Podfile.
+#   - Parameter `version`: bow's version which update Podfile.
+#   - Parameter `branch`: bow's branch which update Podfile.
```
```suggestion
#   - Parameter `branch`: Bow's branch which update Podfile.
```

truizlop reviewed (changes requested):
I have run `nef playground` with no arguments after a fresh install and it fails with the following error:

```
Installing Playground (BowPlayground)... ✅
Updating bow  ❌
[!] error: can not find file '/Users/tomasruizlopez/BowPlayground/Podfile'
```

truizlop committed:
Update bin/nef-playground

Co-Authored-By: miguelangel-dev <madiaz.dev@gmail.com>

truizlop committed:
Update bin/nef-playground

Co-Authored-By: miguelangel-dev <madiaz.dev@gmail.com>

truizlop committed:
Update bin/nef-playground

Co-Authored-By: miguelangel-dev <madiaz.dev@gmail.com>

truizlop committed:
Update bin/nef-playground

Co-Authored-By: miguelangel-dev <madiaz.dev@gmail.com>

miguelangel-dev committed:
update to support parameters in nef-playground and jekyll

miguelangel-dev committed:
fix nef-jekyll setup

miguelangel-dev committed:
refactor name bow -> Bow - keep branding

miguelangel-dev committed:
format menu

miguelangel-dev committed:
common approach to generate playground

miguelangel-dev committed:
update options to launch scripts

miguelangel-dev committed:
added checking for cocoapods version

miguelangel-dev committed:
keep consistency with readme

miguelangel-dev committed:
minor changes

miguelangel-dev committed:
sort menu help

miguelangel-dev committed:
Improve error information in nef-playground

More specific in error when you can not build playground because another project with the same name already exist.

miguelangel-dev committed:
Pods install: run pod repo update before doing pod install

miguelangel-dev committed:
update path for nef-jekyll

miguelangel-dev committed:
fix jekyll path

miguelangel-dev committed:
link the whole playground references when install nef

miguelangel-dev committed:
fix (project | site | main page) path

miguelangel-dev committed:
minor changes
